tests/test_arcaflow_plugin_pcp.py

```diff
@@ -146,6 +146,46 @@ def test_functional_flat(self):
         self.assertEqual("success", output_id)
         self.assertIsInstance(output_data.pcp_output, list)
 
+    def test_functional_user_pmlogger_conf(self):
+        f = open("tests/pmlogger.conf", "r")
+        pmlogger_conf = f.read()
+        input = pcp_plugin.PcpInputParams(
+            pmlogger_interval=1.0,
+            pmlogger_metrics="kernel.all.cpu.user mem.util.used",
+            timeout=3,
+            pmlogger_conf=pmlogger_conf,
+        )
+
+        output_id, output_data = pcp_plugin.StartPcpStep.start_pcp(
+            params=input, run_id="ci_pcp"
+        )
+
+        print(f"==>> output_id is {output_id}")
+        print(f"==>> output_data is {output_data}")
+
+        self.assertEqual("success", output_id)
+        self.assertIsInstance(output_data.pcp_output, list)
+
+    def test_functional_user_pmrep_conf(self):
+        f = open("tests/pmrep.conf", "r")
+        pmrep_conf = f.read()
+        input = pcp_plugin.PcpInputParams(
+            pmlogger_interval=1.0,
+            pmlogger_metrics="kernel.all.cpu.user mem.util.used",
+            timeout=3,
+            pmrep_conf=pmrep_conf,
+        )
```

Comment:
This function seems to be identical to `test_functional_user_pmlogger_conf()` except for the configuration file.  If that's true, consider combining the two functions and driving them with a table.

Reply:
```
import pytest
[...]

    @pytest.mark.parametrize(conf, ("tests/pmlogger.conf", "tests/pmlogger.conf")
    def test_functional_user_pmlogger_conf(self, conf):
        pmlogger_conf = Path(conf).read_text()
        input = pcp_plugin.PcpInputParams(
            pmlogger_interval=1.0,
            pmlogger_metrics="kernel.all.cpu.user mem.util.used",
            timeout=3,
            pmlogger_conf=pmlogger_conf,
        )

        output_id, output_data = pcp_plugin.StartPcpStep.start_pcp(
            params=input, run_id="ci_pcp"
        )

        print(f"==>> output_id is {output_id}")
        print(f"==>> output_data is {output_data}")

        self.assertEqual("success", output_id)
        self.assertIsInstance(output_data.pcp_output, list)
```

Reply:
Dave's answer would be excellent, _if_ we were using the PyTest framework (and if there were a handy way to access the correct parameter key value), but we're instead using `unittest`, which Dave and I have far less experience with.  Nevertheless, the basic idea is the same, and I _think_ the applicable `unittest` concept is the [`subTest()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest) ([docs example](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests)).

So, perhaps something like this:
```python
    def test_functional_user_conf(self):
        configs = [
            { pmlogger_conf: Path("tests/pmlogger.conf").read_text() },
            { pmrep_conf: Path("tests/pmrep.conf").read_text() },
        ]

        for config in configs:
            with self.subTest(config=config):
                input = pcp_plugin.PcpInputParams(
                        pmlogger_interval=1.0,
                        pmlogger_metrics="kernel.all.cpu.user mem.util.used",
                        timeout=3,
                        **config,
                )

                output_id, output_data = pcp_plugin.StartPcpStep.start_pcp(
                    params=input, run_id="ci_pcp"
                )

                print(f"==>> output_id is {output_id}")
                print(f"==>> output_data is {output_data}")

                self.assertEqual("success", output_id)
                plugin.test_object_serialization(pcp_plugin.PerfOutput(output_data.pcp_output))
```
That is, each test scenario is a dictionary containing the key/value pair to pass to the `PcpInputParams()` call, and the scenarios are grouped into a list so that we can iterate over them conveniently.  Because the entries are `dict`'s, we can use the `**` syntax to expand them into keyword parameters in the function call.  Thanks to the `subTest()` context, each iteration will be treated as a separate test by `unittest` (in particular, if the first iteration fails, the second one will still be run, and both results will be reported).

Reply:
> Dave's answer would be excellent, if we were using the PyTest framework

Ah; I almost considered that, but then didn't. 😆 "Some assembly required"...